validate/validate.go

```diff
@@ -490,6 +499,30 @@ func namespaceValid(ns rspec.Namespace) bool {
 	return true
 }
 
+func nestedValid(os, pathA, pathB string) bool {
+	var purePathA, purePathB string
+	if os == "windows" {
+		purePathA = strings.TrimSuffix(pathA, "\\")
+		purePathB = strings.TrimSuffix(pathB, "\\")
+	} else {
+		purePathA = strings.TrimSuffix(pathA, "/")
+		purePathB = strings.TrimSuffix(pathB, "/")
+	}
+
+	if purePathA == purePathB {
+		return false
+	} else if strings.HasPrefix(purePathB, purePathA) {
+		splitedPaths := strings.SplitN(purePathB, purePathA, 2)
+		if os == "windows" && strings.HasPrefix(splitedPaths[1], "\\") {
+			return true
+		} else if os != "windows" && strings.HasPrefix(splitedPaths[1], "/") {
+			return true
+		}
```

Comment:
Actually, the function should probably be `ancestorPath(os string, pathA string, pathB string, strict boolean) (isAncestor boolean)`, where when `strict` is false, `isAncestor` is true when `pathA` and `pathB` are equivalent or `pathB` is a strict ancestor of `pathA`.